### ISSUE TITLE
Broken rocky#balance?

### ISSUE THREAD

**Issue by kacperzuk** [h2non/rocky#78]
I'm trying to make rocky#balance work, but even [the example](https://github.com/h2non/rocky/blob/master/examples/balancer.js) seems to work incorrectly. It returns:

```
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
Status: 200 from server: 1
```

While I expected something more like:

```
Status: 200 from server: 1
Status: 200 from server: 2
Status: 200 from server: 3
Status: 200 from server: 1
Status: 200 from server: 2
Status: 200 from server: 3
Status: 200 from server: 1
Status: 200 from server: 2
Status: 200 from server: 3
Status: 200 from server: 1
```

When I take the first server down, then it's even worse, as I only get `Status: 502 from server: rocky`, it never tries other (working) servers. 

Is that an expected behavior?

**Comment by h2non** [comment]
That behavior is not expected. The issue was introduced in the latest release. I'll fix it asap.

Thank you for reporting.

**Comment by kacperzuk** [comment]
Here's my attempt of fixing this: https://github.com/kacperzuk/rocky/commit/2f82c6602e7dfa1fdcccc26e69055d888294ab03 .

I'm not sure it's the best way, but it passes tests and fixes the problem for me. I'll gladly make a PR if you think it's good enough.

**Comment by h2non** [comment]
Thanks for the proposal.

This is now fixed and properly tested in `rocky@0.4.6`.
Feel free to update your dependency tree:
```bash
npm i rocky --force
```
